modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -723,6 +725,25 @@ private void initCachesOnLocalJoin() throws IgniteCheckedException {
         }
 
         cctx.cache().startCachesOnLocalJoin(caches, initialVersion());
+
+        ensureClientCachesStarted();
+    }
+
+    /**
+     * Start client caches if absent.
+     */
+    private void ensureClientCachesStarted() {
+        GridCacheProcessor cacheProcessor = cctx.cache();
+        List<CacheConfiguration> notStartedCacheConfigs = new ArrayList<>();
+        for (CacheConfiguration cCfg : cctx.gridConfig().getCacheConfiguration()) {
+            if (!cacheProcessor.cacheNames().contains(cCfg.getName())) {
+                notStartedCacheConfigs.add(cCfg);
+            }
+        }
+
+        if (!notStartedCacheConfigs.isEmpty()) {
+            cacheProcessor.dynamicStartCaches(notStartedCacheConfigs, false, false);
+        }
```

Comment:
As per Ignite conventions there shouldn't be curly braces {} if there's only one statement under if ()